### PR TITLE
Reflection and clear coat

### PR DESCRIPTION
**Pull request by anisunity**
### Purpose of this PR
When a material has a clear coat layer, we want to use the clear coat's roughness to evaluate SSR or Raytraced reflections.

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Automated Tests**: Added an automatic test 2801_SSR that tests SSR and the new feature for SSR

### CODE REVIEW

**Comment by sebastienlagarde** [comment]
Please katana link :)

**Comment by anisunity** [comment]
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2FReflectionClearCoat&automation-tools_branch=add-platform-filter&unity_branch=trunk

**Comment by anisunity** [comment]
tests are failing
I am gonna look why

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 1]
```diff
@@ -2844,6 +2844,9 @@ void RenderSSR(HDCamera hdCamera, CommandBuffer cmd, ScriptableRenderContext ren
                 int w = hdCamera.actualWidth;
                 int h = hdCamera.actualHeight;
 
+                // Evaluate the clear coat mask texture based on the lit shader mode
+                RenderTargetIdentifier clearCoatMask = hdCamera.frameSettings.litShaderMode == LitShaderMode.Deferred ? m_GbufferManager.GetBuffersRTI()[2] : Texture2D.blackTexture;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: It miss a clear of the GBuffer2 to be able to handle all the others forward material

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Question, is Texture2D.blackTexture a RGBA black texture or a RGB black texture (so in this case black is 1)

**Pull request merged**